conans/cli/api/subapi/new.py

```diff
@@ -0,0 +1,78 @@
+import fnmatch
+import os
+
+from jinja2 import Template
+
+from conans.cli.api.helpers.new import new_templates
+from conans.cli.api.subapi import api_method
+from conans.cli.conan_app import ConanApp
+from conans.errors import ConanException
+from conans.util.files import load
+from conans import __version__
+
+
+class NewAPI:
+
+    _NOT_TEMPLATES = "not_templates"
+
+    def __init__(self, conan_api):
+        self.conan_api = conan_api
+
+    @api_method
+    def new(self, template, definitions):
+        app = ConanApp(self.conan_api.cache_folder)
+
+        # Find the template
```

Comment:
All these lines about finding the template shouldn't be here. API methods should be more "atomic" regarding the arguments. Probably this function should receive the list of file contents directly? Then have other/s to "scan" them (to the cache directory or the predefined one): `scan_cache()`, `get_preset_files()`, `scan_folder()`
Also, the _NOT_TEMPLATES doesn't sound good, (and not tested?), probably can be replaced with a more explicit argument with excluded patterns in the `scan_folder()`?
So, the `def new(self, template, definitions):` might be more like  `def render(self, files, definitions):` and then 
 it would be almost a loop of jinja renders.

Reply:
But we would still move the logic to another ``NewAPI`` method, right? It would be a bit challenging to implement the logic in the command side only, there is a lot of business logic in it. So I agree to split this in 2 methods: 1 for locating the files, and another for rendering.

The problem with ``not_templates`` is that those are files that should still be copied, they need to be loaded and saved, they are part of the template. Just they don't need to be passed through Jinja, that can crash or corrupt the file. I will add testing for it.

Reply:
There is already a test for ``not_templates``, it is called: ``test_template_image_files``. Test extracted from user issue: https://github.com/conan-io/docs/issues/2165

Reply:
Done, please check.

Reply:
`get_template_files` is still doing too many things. I still think that there are three potential methods there: `scan_cache_templates(), get_preset_template_files(), get_template_files(FOLDER)`, or maybe only 2 if we provide the path to the cache folder templates: `get_preset_files(), get_template_files(FOLDER)` + `.template_cache_folder`